test/Table_Internal_Tests/src/Types/Table_Column_Any_Spec.enso

```diff
@@ -0,0 +1,35 @@
+from Standard.Base import all
+from Standard.Table import all
+
+import Standard.Base.Internal.Meta_Helpers
+
+from Standard.Test import all
+
+# using Meta_Helpers requires us to run with --disabled-private-checks option
+# hence this module is in an "_Internal_" test suite
+private assert_type_name value:Any expected_name:Text = Frame_Hider.hide <|
+    actual_name = Meta_Helpers.simple_type_name_builtin value
+    actual_name . should_equal expected_name
+
+add_specs suite_builder =
+    suite_builder.group "Intersection types in table" group_builder->
+        group_builder.specify "Table of Integer" <|
```

Comment:
- this test puts the [requirements](https://github.com/enso-org/enso/issues/13390#issuecomment-3118545211) into code
- done in a7ff612b092ebfe4da7a506352ddc1ec3e6c50f3
- @jdunkerley please check and adjust to match exactly your expectations